### DISCUSSION TITLE
can rxjs websocket send ping or pong?

### DISCUSSION THREAD

**Discussion by bongjinpark1** [ReactiveX/rxjs#7456] [Q&A]
```javascript
import { webSocket } from 'rxjs/websocket'

const ws = webSocket({
  url: 'ws://localhost:3000',
  openObserver: {
    next: () => {
      ws.next(0x89) // <- can i send ping frame? this is not working.
    }
  }
})

ws.subscribe()
```

hello.

I want to check websocket connection status by pingpong heartbeat.

but, I can't found how to send ping/pong frame.

Is there any interfaces for it?

Thanks.

**Comment by demensky** [comment]
The browser does not provide any methods that would allow you to send a ping/pong frame. As a result, RxJS also cannot send ping/pong.

The only option is to implement your own ping/pong on top of regular messages.
